mne/channels/montage.py

```diff
@@ -1292,40 +1345,49 @@ def _set_montage(info, montage, update_ch_names=False, set_dig=True):
                  'left untouched.')
 
     elif isinstance(montage, DigMontage):
+        _mnt = montage if montage._coord_frame == 'head' else transform_to_head(montage.copy())
+        _raise = (  # XXX: deprecated to remove in 0.20
+            True if raise_if_subset is DEPRECATED_PARAM else raise_if_subset
+        )
+
+        def _backcompat_value(pos, ref_pos):
+            if any(np.isnan(pos)):
+                return np.full(6, np.nan)
+            else:
+                return np.concatenate((pos, ref_pos))
+
+        ch_pos = _mnt._get_ch_pos()
+        eeg_ref_pos = ch_pos.pop('EEG000', np.zeros(3))
+
+        # This raises based on info being sub/supper set of montage
+        _pick_chs = partial(
+            pick_types, exclude=[], eeg=True, seeg=True, ecog=True, meg=False,
+        )
```

Comment:
imagine I have EEG locations already present in info and SEEG channels. I want to set the SEEG channel locations while not changing the EEG locations. I think this would break.

Reply:
In that case, you need to tinker your montages. We decided that when you were
setting a montage you set the whole thing.
We said it was not making sense to set up some channels but let others unset and
just warn. That this was evil.

I would keep it restrictive. And if you want to do custom stuff, you create your
custom data structure (aka `my_montage`) and set it.

Maybe I do not fully understand the use case. 

But this is how I would set something that has eeg and seeg:
```py
raw = read_raw_bla() # with some EEG (from template) and SEEG (from surgery)

my_montage = make_standard_montage('foo') + make_dig_montage(ch_pos_of_seeg)
raw.set_montage(my_montage)
```

And if for whatever reason setting the montage is not something that you do at
once right after the reading and its something that needs changing during your
pipeline (I don't see why, but just lets pretend), I would write it like this:

```py
raw = read_raw_bla(..) # with some EEG (from template) and SEEG (from surgery)
original_montage = read_dig_xxx(..)
raw.set_montage(original_montage)

# do stuff

# now we need to modify some part of the locations

seeg_montage = make_dig_montage(ch_pos_of_seeg)
picks = mne.pick_type(raw.info, meg=False, eeg=True)
eeg_montage = make_dig_montage(ch_pos=dict(zip(
    [raw.info['ch_names'][ii] for ii in picks],
    [raw.info['chs'][ii]['loc'][:3] for ii in picks],
)))

raw.set_montage(eeg_montage + seeg_montage)
```

IMHO doing something like this is likely to bring problems:
```py
raw = read_raw_bla() # with some EEG (from template) and SEEG (from surgery)

eeg_montage = make_standard_montage('foo')
seeg_montage = make_dig_montage(ch_pos_of_seeg)
raw.set_montage(eeg_montage, eeg=True, seeg=False, foo=False)
raw.set_montage(eeg_montage, types=['eeg', 'seeg'])
```

```py
raw = read_raw_bla() # with some EEG (from template)

eeg_montage = make_standard_montage('foo')
seeg_montage = make_dig_montage(ch_pos_of_seeg)
raw.set_montage(eeg_montage + seeg_montage)  # do not break 'cos we spot that seeg does not apply
```